docs/src/modules/components/MarkdownDocs.js

```diff
@@ -70,7 +71,13 @@ function MarkdownDocs(props) {
           }
 
           if (renderedMarkdownOrDemo.component) {
-            const Component = markdownComponents[renderedMarkdownOrDemo.component];
+            const name = renderedMarkdownOrDemo.component;
+            const Component = componentComponents?.[name];
+
+            if (Component === undefined) {
+              throw new Error('Missing component');
+            }
+
             return (
               <Wrapper key={index} {...(isJoy && { mode: theme.palette.mode })}>
                 <Component headers={headers} options={renderedMarkdownOrDemo} />
```

Comment:
I think we can update the passing props to something like this:

```js
<Component
  {...renderedMarkdownOrDemo}
  markdown={docs[userLanguage] || docs.en}
/>
```

all of the data in `{{ ...data }}` in the markdown is spread to the component and the information about the markdown file is consolidated in the `markdown` prop.

```md
// example
---
title: test
---

## Test

{{ "component": "TestComponent.js", "design": false }}
```

The component will get the props:
```js
function ComponentLinkHeader({ markdown, design }) {
  const { headers } = markdown;
  return (...)
}
```

Reply:
We can

And would you group everything under `props` in the markdown ?

```md
{{ "component": "TestComponent.js", "props": { "design": false } }}
```

As you proposed in https://github.com/mui/material-ui/pull/34243#issuecomment-1243313108

Reply:
I don't think so, we can just spread the props like you did.